crates/core/src/repository.rs

```diff
@@ -416,8 +416,28 @@ impl<P, S> Repository<P, S> {
     /// [`RepositoryErrorKind::ListingRepositoryConfigFileFailed`]: crate::error::RepositoryErrorKind::ListingRepositoryConfigFileFailed
     /// [`RepositoryErrorKind::MoreThanOneRepositoryConfig`]: crate::error::RepositoryErrorKind::MoreThanOneRepositoryConfig
     pub fn config_id(&self) -> RusticResult<Option<ConfigId>> {
-        let config_ids = self
-            .be
+        return self.config_id_with_backend(&self.be);
+    }
+
+    /// Returns the Id of the config file
```

Comment:
to avoid confusion with `pub fn config_id(&self) -> ...` 
you could change the description "Returns the Id of the config file corresponding to a specific backend."